rustls-platform-verifier/src/verification/others.rs

```diff
@@ -169,15 +168,52 @@ impl ServerCertVerifier for Verifier {
                 e
             })
     }
+
+    fn verify_tls12_signature(
+        &self,
+        message: &[u8],
+        cert: &pki_types::CertificateDer<'_>,
+        dss: &DigitallySignedStruct,
+    ) -> Result<HandshakeSignatureValid, TlsError> {
+        self.inner
+            .get_or_try_init(|| self.init_verifier())?
+            .verify_tls12_signature(message, cert, dss)
+    }
+
+    fn verify_tls13_signature(
+        &self,
+        message: &[u8],
+        cert: &pki_types::CertificateDer<'_>,
+        dss: &DigitallySignedStruct,
+    ) -> Result<HandshakeSignatureValid, TlsError> {
+        self.inner
+            .get_or_try_init(|| self.init_verifier())?
+            .verify_tls13_signature(message, cert, dss)
+    }
+
+    fn supported_verify_schemes(&self) -> Vec<SignatureScheme> {
+        let verifier = match self.inner.get_or_try_init(|| self.init_verifier()) {
+            Ok(v) => v,
+            Err(_) => return Vec::default(),
+        };
+        verifier.supported_verify_schemes()
+    }
+}
```

Comment:
This trait fn doesn't support returning a `Result` so if the once cell verifier init fails it returns an empty vec. Alternatively we could maintain a `CryptoProvider` in the other verifier struct to use instead of the once-cell wrapped webpki verifier. I thought this approach was better in sum.

Reply:
I think this approach seems reasonable to me. Now that we have 4 places calling `get_or_try_init`, I wonder if a helper function on `&self`, something like `get_or_init_verifier(), would be useful to abstract over the `OnceCell` boilerplate (`get_or_try_init(|| self.init_verifier()))` more.

Reply:
Sounds reasonable, I can tack that on

Reply:
Fixed.